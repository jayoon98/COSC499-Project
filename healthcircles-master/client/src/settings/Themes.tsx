import React, { useContext } from 'react';
import { Text, StyleSheet, View } from 'react-native';
import { Colors } from '../colors';
import { ThemeContext, themes, Theme } from '../common/ThemeContext';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { saveUserTheme } from '../services/theme';

const domains = ['social', 'emotional', 'physical', 'mental', 'spiritual'];

type ThemeCardProps = {
  themeKey: string;
  theme: Theme;
  selectedTheme: Theme;
  onSelect: (theme: Theme) => void;
};

function ThemeCard({
  themeKey,
  theme,
  selectedTheme,
  onSelect,
}: ThemeCardProps) {
  const makeDots = () => {
    return domains.map((d, i) => {
      const style = {
        ...styles.domainCircle,
        backgroundColor: theme[d],
      };
      return <View key={i} style={style} />;
    });
  };

  return (
    <TouchableOpacity
      onPress={(e) => {
        onSelect(theme);
        saveUserTheme(themeKey);
      }}
    >
      <View style={styles.themeCard}>
        <View style={{ marginRight: 12 }}>
          {theme.name === selectedTheme.name ? (
            <FontAwesomeIcon icon={faCheck} />
          ) : (
            <View style={{ width: 16, height: 16 }}></View>
          )}
        </View>
        <Text style={{ width: 96 }}>{theme.name}</Text>
        {makeDots()}
      </View>
    </TouchableOpacity>
  );
}

export function Themes() {
  const { theme, setTheme } = useContext(ThemeContext);

  return (
    <View style={styles.themeSelection}>
      <View style={styles.themesList}>
        {Object.keys(themes).map((k, i) => (
          <ThemeCard
            themeKey={k}
            theme={themes[k]}
            selectedTheme={theme}
            key={i}
            onSelect={(t) => setTheme(t)}
          />
        ))}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  themeSelection: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    width: '100%',
  },
  themesList: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
  },
  themeCard: {
    padding: 18,
    borderRadius: 14,
    margin: 12,
    marginTop: 0,
    backgroundColor: Colors.background,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 5.6,
    elevation: 5,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  domainCircle: {
    height: 32,
    width: 32,
    borderRadius: 16,
    backgroundColor: Colors.offwhite,
    marginRight: 10,
  },
  header: {
    height: 100,
    display: 'flex',
    justifyContent: 'flex-end',
    padding: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#f2f6fa',
  },
});
