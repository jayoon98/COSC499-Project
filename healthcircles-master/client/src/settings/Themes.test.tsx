import React from 'react';
import renderer from 'react-test-renderer';
import { Themes } from './Themes';

jest.mock('@react-native-picker/picker', () => ({
  Picker: '',
}));

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

jest.mock('@react-native-community/async-storage', () => ({
  AsyncStorage: '',
}));

jest.mock('rn-tourguide', () => ({
  TourGuideProvider: '',
}));

it('renders correctly', () => {
  const tree = renderer.create(<Themes />).toJSON();
  expect(tree).toMatchSnapshot();
});
