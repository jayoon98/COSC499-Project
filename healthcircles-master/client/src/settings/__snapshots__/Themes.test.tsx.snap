// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "elevation": 5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 12,
              "marginTop": 0,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 5.6,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "width": 16,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "width": 96,
              }
            }
          >
            Theme 1
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#fa7f72",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#389393",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#5350ef",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#f5a25d",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#a8dda8",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "elevation": 5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 12,
              "marginTop": 0,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 5.6,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "width": 16,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "width": 96,
              }
            }
          >
            Theme 2
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#9cb380",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#522a27",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#7785ac",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#c73e1d",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#c59849",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "elevation": 5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 12,
              "marginTop": 0,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 5.6,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "width": 16,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "width": 96,
              }
            }
          >
            Theme 3
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#04395e",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#70a288",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#031d44",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#dab785",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#d5896f",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "elevation": 5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 12,
              "marginTop": 0,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 5.6,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "width": 16,
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "width": 96,
              }
            }
          >
            Theme 4
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#f5efa6",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#BDEFD0",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#ff99c8",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#a9def9",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#e4c1f9",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "elevation": 5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 12,
              "marginTop": 0,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 5.6,
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 12,
              }
            }
          >
            <
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f00c",
                    "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
                  ],
                  "iconName": "check",
                  "prefix": "fas",
                }
              }
            />
          </View>
          <Text
            style={
              Object {
                "width": 96,
              }
            }
          >
            Theme 5
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#00897b",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#ffa000",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#303f9f",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#c6282b",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#673ab7",
                "borderRadius": 16,
                "height": 32,
                "marginRight": 10,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
